core/src/weighted_shuffle.rs

```diff
@@ -120,9 +124,8 @@ mod tests {
 
     #[test]
     fn test_weighted_best() {
-        let mut weights = vec![(std::u32::MAX as u64, 0); 3];
-        weights.push((1, 5));
+        let weights = vec![(100, 0), (1000, 1), (10_000, 2), (10, 3)];
         let best = weighted_best(&weights, ChaChaRng::from_seed([0x5b; 32]));
-        assert_eq!(best, 5);
+        assert_eq!(best, 2);
```

Comment:
Between the test data and the ambiguous function name, it's not obvious that 2 is referring to index 2 and not the 2 at `weights[2].1`.